### PR TITLE
PTX-2580 Refactor backup tests

### PR DESCRIPTION
**Pull request by stgleb**

### CODE REVIEW

**Comment by stgleb** [comment]
Jobs run:

**Comment by vinayakshnd** [comment]
@stgleb there are some linting errors. Can u fix those? These errors can be seen by running `make all` command in torpedo root dir.

```
./tests/backup/backup_util.go:29:2: should not use dot imports
./tests/backup/backup_util.go:56:1: exported function SetupBackup should have comment or be unexported
./tests/backup/backup_util.go:76:1: exported function TearDownBackupRestore should have comment or be unexported
./tests/backup/backup_util.go:165:1: exported function CreateBucket should have comment or be unexported
./tests/backup/backup_util.go:175:1: exported function CreateS3Bucket should have comment or be unexported
./tests/backup/backup_util.go:209:1: exported function CreateAzureBucket should have comment or be unexported
./tests/backup/backup_util.go:232:1: exported function DeleteBucket should have comment or be unexported
./tests/backup/backup_util.go:242:1: exported function DeleteS3Bucket should have comment or be unexported
./tests/backup/backup_util.go:281:1: exported function DeleteAzureBucket should have comment or be unexported
./tests/backup/backup_util.go:449:1: exported function CreateBackupLocation should have comment or be unexported
./tests/backup/backup_util.go:629:1: exported function DumpKubeconfigs should have comment or be unexported
./tests/backup/backup_util.go:735:1: exported function GetNodesForBackup should have comment or be unexported
make: *** [lint] Error 1
```

**Comment by vinayakshnd** [comment]
@stgleb 

> Jobs run:
> https://jenkins.portworx.co/job/Torpedo/job/tp-eks-backup/248/
> https://jenkins.portworx.co/job/Torpedo/job/tp-aks-backup/263/

above jobs have not run backup tests. Can u pls check them

**Comment by stgleb** [comment]
> @stgleb
> 
> > Jobs run:
> > https://jenkins.portworx.co/job/Torpedo/job/tp-eks-backup/248/
> > https://jenkins.portworx.co/job/Torpedo/job/tp-aks-backup/263/
> 
> above jobs have not run backup tests. Can u pls check them

fixed linting

**Comment by vinayakshnd** [comment]
No need for this check: https://github.com/portworx/torpedo/pull/502/files#diff-3ca1d6fd1651a59da2ec30fed2263e9eR280
The next line is anyways checking if the array is empty or not.

File: tests/backup/backup_util.go [hunk 1]
```diff
@@ -0,0 +1,865 @@
...
+				&appsapi.Deployment{
+					ObjectMeta: meta_v1.ObjectMeta{
+						Name:      storkDeploymentName,
+						Namespace: storkDeploymentNamespace,
+					},
+				},
+			},
+		},
+	}
+	logrus.Infof("Execute task for killing stork")
```
Comment (vinayakshnd) [inline tests/backup/backup_util.go:143]: Log statement Not required

File: tests/backup/backup_util.go [hunk 2]
```diff
@@ -0,0 +1,865 @@
...
+	err := dumpKubeConfigs(configMapName, kubeconfigList)
+
+	if err != nil {
+		return fmt.Errorf("failed to get kubeconfigs [%v] from configmap [%s]", kubeconfigList, configMapName)
+	}
+	return nil
+}
+
+// CreateProviderClusterObject creates cluster for each cluster per each cloud provider
+func CreateProviderClusterObject(provider string, kubeconfigList []string, cloudCred, orgID, clusterName string) error {
```
Comment (vinayakshnd) [inline tests/backup/backup_util.go:651]: Is this function being used anywhere?

**Comment by stgleb** [comment]
> No need for this check: https://github.com/stgleb/torpedo/blob/PTX-2589/tests/backup/backup_test.go#L280
> The next line is anyways checking if the array is empty or not.
> Same Here: https://github.com/stgleb/torpedo/blob/PTX-2589/tests/backup/backup_test.go#L287
> 
> Log statement not required: https://github.com/stgleb/torpedo/blob/PTX-2589/tests/backup/backup_test.go#L368
> 
> ctx.Done() can be replaced with WaitGroup(): https://golang.org/pkg/sync/#WaitGroup

Done

File: tests/backup/backup_test.go [hunk 3]
```diff
@@ -162,20 +83,28 @@ var _ = Describe("{BackupCreateKillStorkRestore}", func() {
 
 	labelSelectores := make(map[string]string)
 	namespaceMapping = make(map[string]string)
-
+	provider, err := getProvider()
+	Expect(err).NotTo(HaveOccurred(),
+		fmt.Sprintf("Failedto get provider. Error: [%v]", err))
```
[the remark this answers is not in the document]
Comment (stgleb) [reply tests/backup/backup_test.go]: Done

File: tests/backup/backup_test.go [hunk 4]
```diff
@@ -208,9 +137,11 @@ var _ = Describe("{BackupCreateKillStorkRestore}", func() {
 			backupName := fmt.Sprintf("%s-%s", backupNamePrefix, namespace)
 			Step(fmt.Sprintf("Create backup full name %s:%s:%s",
 				sourceClusterName, namespace, backupName), func() {
-				CreateBackup(backupName,
+				err = createBackup(backupName,
 					sourceClusterName, backupLocationName,
 					[]string{namespace}, labelSelectores, orgID)
+				Expect(err).NotTo(HaveOccurred(),
+					fmt.Sprintf("Failed dekete backup %s Error: [%v]", backupName, err))
```
[the remark this answers is not in the document]
Comment (stgleb) [reply tests/backup/backup_test.go]: Done

File: tests/backup/backup_test.go [hunk 5]
```diff
@@ -504,8 +421,9 @@ var _ = Describe("{MultiProviderBackupKillStork}", func() {
 					providerClusterConfigPath, err := getProviderClusterConfigPath(provider, kubeconfigList)
```
[the remark this answers is not in the document]
Comment (stgleb) [reply tests/backup/backup_test.go]: Done

File: tests/backup/backup_test.go [hunk 6]
```diff
@@ -395,7 +318,9 @@ var _ = Describe("{MultiProviderBackupKillStork}", func() {
 					Expect(err).NotTo(HaveOccurred(),
 						fmt.Sprintf("Failed to get kubeconfig path for provider %s cluster. Error: [%v]", provider, err))
 					logrus.Infof("Set context to %s", providerClusterConfigPath)
```
[the remark this answers is not in the document]
Comment (stgleb) [reply tests/backup/backup_test.go]: Done

File: tests/backup/backup_test.go [hunk 7]
```diff
@@ -561,8 +483,9 @@ var _ = Describe("{MultiProviderBackupKillStork}", func() {
 					Expect(err).NotTo(HaveOccurred(),
 						fmt.Sprintf("Failed to get kubeconfig path for provider %s cluster. Error: [%v]", provider, err))
 					logrus.Infof("Set config to %s", providerClusterConfigPath)
```
[the remark this answers is not in the document]
Comment (stgleb) [reply tests/backup/backup_test.go:461]: Done

File: tests/backup/backup_util.go (hunk 1 above)
Comment (stgleb) [reply tests/backup/backup_util.go]: Done

File: tests/backup/backup_util.go (hunk 2 above)
Comment (stgleb) [reply tests/backup/backup_util.go]: Done

**Pull request merged**